lib/galaxy/webapps/galaxy/services/workflows.py

```diff
@@ -124,6 +124,8 @@ def invoke_workflow(
         workflow_id,
         payload: InvokeWorkflowPayload,
     ) -> Union[WorkflowInvocationResponse, List[WorkflowInvocationResponse]]:
+        if trans.anonymous:
```

Comment:
That's different from inactive users, which would be needed to close out https://github.com/galaxyproject/galaxy/issues/18147

Reply:
If we only check for user activation (`trans.check_user_activation()`) in that line, we still create errored datasets and execution continues beyond that line, because:
```
@property
def user_is_active(self) -> bool:
    return not self.app.config.user_activation_on or self.user is None or self.user.active
```
... and this property considers anonymous users as `active`. Here is an example:

https://github.com/galaxyproject/galaxy/assets/78516064/b868935c-4ad1-4276-9115-97a300413b0e

If we instead check for `trans.anonymous` (like I have it rn), we immediately throw an error and we never get to the point that created this ticket. For e.g.:

https://github.com/galaxyproject/galaxy/assets/78516064/1953334e-e609-45a2-ba47-278f1251d737

_(Keep in mind that this workflow run form route is inaccessible for anon users, I just made it accessible for demonstrative purposes)_

If we still want to prevent inactive users AND anonymous users BOTH from invoking a workflow, I can use another check instead of `trans.check_user_activation()` which checks for:
```
return self.user is not None and (not self.app.config.user_activation_on or self.user.active)
```